### PR TITLE
[ai] Add client info to headers for OpenAI classes

### PR DESCRIPTION
**Pull request by diondrapeck**
# Description

This PR adds a header containing the client user agent `<package name>/<version>` to every OpenAI client initialized so that calls originating from the SDK can be distinguished in telemetry.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by luigiw — APPROVED** [review]

File: sdk/ai/azure-ai-generative/azure/ai/generative/_user_agent.py [hunk 1]
```diff
@@ -4,3 +4,4 @@
 from azure.ai.generative._version import VERSION
 
 USER_AGENT = "{}/{}".format("azure-ai-generative", VERSION)
+OPENAI_USER_AGENT = f"AzureOpenAI/Python 1.7.1 {USER_AGENT}"
```
Comment (singankit) [inline sdk/ai/azure-ai-generative/azure/ai/generative/_user_agent.py:7]: 1.7.1 is this AzureOpenAI SDK version ?

File: sdk/ai/azure-ai-resources/azure/ai/resources/_index/_embeddings/openai.py [hunk 2]
```diff
@@ -76,11 +77,13 @@ def __init__(
                     api_key=self.api_key,
                     api_version=self.api_version,
                     azure_endpoint=self.api_base,
+                    default_headers={"Client-Type-User-Agent": OPENAI_USER_AGENT},
```
Comment (singankit) [inline sdk/ai/azure-ai-resources/azure/ai/resources/_index/_embeddings/openai.py:80]: Lets make a constant for it ?

Comment (diondrapeck) [reply sdk/ai/azure-ai-resources/azure/ai/resources/_index/_embeddings/openai.py:80]: for "Client-Type-User-Agent"?

File: sdk/ai/azure-ai-generative/azure/ai/generative/_user_agent.py (hunk 1 above)
Comment (diondrapeck) [reply sdk/ai/azure-ai-generative/azure/ai/generative/_user_agent.py:7]: Yes, but gonna remove this constant. I'm going to only pass our user agent.

File: sdk/ai/azure-ai-resources/azure/ai/resources/_index/_embeddings/openai.py (hunk 2 above)
Comment (singankit) [reply sdk/ai/azure-ai-resources/azure/ai/resources/_index/_embeddings/openai.py]: Yes

**Review by singankit — APPROVED** [review]

**Pull request merged**